src/game/resource.h

```diff
@@ -34,18 +34,22 @@ typedef enum {
     RESOURCE_IRON,
     RESOURCE_MARBLE,
     RESOURCE_GOLD,
+    RESOURCE_SAND,
```

Comment:
Adding these in the middle of the list will change existing list will shift the IDs of existing resources.
This will break any savegames (requests, events, cart depot orders, warehouse storage data?) that make use of those ids.
I think it is much safer to add the new resources to the end of the list.

Reply:
Actually they won't, I safeguarded against that issue when making the resources dynamic.

Check `resource_set_mapping` and `resource_remap`, which are always called when loading a new save.

Reply:
Are you sure?
If I load any of my old savegames the warehouses all appear empty. In the screenshot there are deliveries waiting to unload because the warehouse SHOULD be full.
![image](https://github.com/Keriew/augustus/assets/2726355/1493e1f7-3af3-4e8b-a045-17e0462e9953)


Also it is definitely not applying to cart depot orders. The depot in question is meant to be wine, not furniture.
![image](https://github.com/Keriew/augustus/assets/2726355/93f13165-569c-4392-9238-43acdb40453e)

The cart pusher of the depot now saying it is moving furniture, but there is still wine in his cart from the savegame.
![image](https://github.com/Keriew/augustus/assets/2726355/1fe01207-cac2-43fe-8102-5d88cc9ad71b)

Reply:
The cart depot is likely on me, since I had not implemented `resource_remap` in `b->data.depot.current_order.resource_type = buffer_read_i8(buf);`

Reply:
Note: Granaries work fine, as well as the contents of cart pusher carts. (As far as I can tell, but I did check multiple old savegames.)
It seems to be specifically warehouses that end up completely empty.
And then the cart depot order that doesn't take the remap into account.

Reply:
Yeah, the remap needs to be added to depots then, but I don't know why warehouses aren't working.

Reply:
That issue is fixed, but now workhops aren't behaving properly (mostly because of my `has_raw_materials` change).

Reply:
Will you update cart depots as part of this PR or should I do that as a follow-up PR?

Reply:
No, I'll do it and also try to fix the remaining bugs.

I'm also thinking about merging your cart depot scrollbar PR to this PR if you don't mind.

Reply:
Not at all. Go ahead.

I'll mark this as resolved for now, since I think this covers the only real 'bug' I found when testing. Everything looks good on the PR now.